docs/planning/otel/OTelWebSdk.md

```diff
@@ -0,0 +1,1567 @@
+# OpenTelemetry Web SDK Specification
+
+## Executive Summary
+
+The OpenTelemetry Web SDK is a modern, enterprise-grade telemetry solution designed specifically for web applications. It provides full OpenTelemetry API compliance while addressing the unique challenges of browser environments and multi-team development scenarios.
+
+**Key Differentiators:**
+- **Multi-Instance Architecture**: Multiple SDK instances can coexist without conflicts, enabling team isolation
+- **Web-Optimized Performance**: Minimal bundle size, superior browser compatibility (ES2020+ with ES5 fallback)
+- **Enterprise Features**: Dynamic configuration, complete cleanup/unload, multi-tenant support
+- **Comprehensive Telemetry**: Full distributed tracing, structured logging, basic metrics, plus web-specific instrumentation
+
+**Primary Use Cases:**
+- Large-scale web applications with multiple development teams
+- Enterprise applications requiring compliance and governance controls  
+- Modern web frameworks (React, Angular, Vue, Next.js, etc.)
+- Progressive Web Apps and Single Page Applications
+- Applications requiring migration from existing telemetry solutions
+
+## Overview
+
+The OpenTelemetry Web SDK is designed as a modern, modular implementation that follows the OpenTelemetry specification while providing enhanced flexibility and performance for web applications. It delivers a complete observability solution encompassing distributed tracing, structured logging, and basic metrics collection through a multi-instance SDK factory that enables multiple teams to efficiently share resources while maintaining isolation. This factory pattern is particularly beneficial for modern web applications where multiple teams or components need isolated telemetry contexts in the same runtime while sharing underlying infrastructure.
+
+## Why a Web-Specific OpenTelemetry SDK?
+
+### **Web Application Requirements & Challenges**
+
+While standard OpenTelemetry SDKs provide excellent observability for backend services, web applications have distinct requirements that justify a specialized implementation:
+
+#### **1. Browser Environment Constraints**
+- **Bundle Size Sensitivity**: Web applications must minimize JavaScript bundle size for performance
+- **Tree-Shaking Requirements**: Dead code elimination is critical for production builds
+- **Browser Compatibility**: Runtime requirement with graceful detection and fallback for unsupported browsers
+  - **Minimum Language Support**: This SDK will target ES2020 features (async/await, optional chaining, nullish coalescing, etc.)
```

Comment:
async / await is problematic -- especially during page unload scenario where we can't do ANYTHING asynchronously (which is the default operation of async / await)... So will need to be used sparingly, or with wrappers